render.go

```diff
@@ -333,15 +340,14 @@ func (r *Render) renderLine(line string, lexer *Lexer, diagnostics []lsp.Diagnos
 			a := strings.SplitAfter(s, v.Text)
 			s = strings.TrimPrefix(s, a[0])
 
-			pos += len(a[0])
-
 			if hasDiagnostic(pos, diagnostics) {
 				r.renderDiagnostic(a[0])
 			} else {
 				r.out.SetColor(v.Color, r.inputBGColor, false)
 				r.out.WriteStr(a[0])
 			}
 
+			pos += len(a[0])
```

Comment:
is there a difference in doing this here instead of above? just curious why this changed

Reply:
Yeah: we call hasDiagnostic with pos and that's why it's important to check if it hasDiagnostic with the pos where we'll print the character (line347) and not the next one. 

The highlighting was basically being shifted by 1 with the line before the if/else

Reply:
aah yeah right, missed that